### PR TITLE
Update go version

### LINKED ISSUE #86: Go version is updated to 1.15
AC:
- [x] Go version is updated to 1.15
- [x] Test automation runs against the current and previous Go version (e.g. not just against a single version) - in this case, Go 1.15 and Go 1.14, to ensure the client works with both Go versions
- [x] The README is clear about which Go versions the client is tested against

### PR DESCRIPTION
**Pull request by BradleyBoutcher**
### What does this PR do?
- Reorganize scripts to improve developer experience 
- Add support for testing multiple golang versions at once
- Updated documentation

### What ticket does this PR close?
Resolves #86 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [X] The changes in this PR do not require tests

#### Documentation
- [X] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -24,18 +21,45 @@ services:
...
     image: registry2.itci.conjur.net/conjur-appliance-cuke-master:4.9-stable
     security_opt:
       - seccomp:unconfined
     ports:
       - 443
 
-  test:
-    container_name: api-go-test
-    build: .
+  test-1.14:
```
Comment (diverdane) [inline docker-compose.yml:30]: The docker-compose definition for `test-1.14` looks almost identical to the definition for `test-1.15` service.
Also, I think the `test-1.14` service is run mutually exclusive from the `test-1.15` service.
So I think you can use docker-compose override, i.e., create a base `test` service in docker-compose
that has default `1.15` values, and create a separate `docker-compose.override-1.14.yml` that simply overrides a few of the values in the base definition.

Just a thought. Not a showstopper.

**Review by diverdane — COMMENTED** [review]
Looks good! a few nit suggestions, but nothing major to stop progress.
Nice work on the dev environment!

File: bin/start-conjur.sh [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  api_key=$(exec_on conjur conjurctl role retrieve-key cucumber:user:admin | tr -d '\r')
+
+  if ! oss_only; then
+    announce "Running cuke setup..."
+    exec_on cuke-master bash -c 'conjur authn login -u admin -p secret'
+    exec_on cuke-master conjur user create --as-group security_admin alice
+    exec_on cuke-master conjur host create --as-group security_admin bob
+    exec_on cuke-master conjur variable create existent-variable-with-undefined-value
+
+    vars=(
```
Comment (diverdane) [inline bin/start-conjur.sh:49]: Maybe add a comment here explaining why these secrets are being added for `cuke-master`? I think they're being added as sample secrets that can be conveniently viewed in the dev environment?

File: docker-compose.yml [hunk 3]
```diff
@@ -24,18 +21,45 @@ services:
       CONJUR_ACCOUNT: cucumber
```
Comment (izgeri) [inline docker-compose.yml:21]: can't comment just above but we should move this to `cyberark/conjur-cli:5`

File: bin/start-conjur.sh [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+main() {
+  # Build test container & start the cluster
+  announce "Pulling images..."
+  if oss_only; then
+      docker-compose -p $COMPOSE_PROJECT_NAME pull conjur
+  else
+      docker-compose -p $COMPOSE_PROJECT_NAME pull conjur cuke-master
+  fi
+  docker-compose -p $COMPOSE_PROJECT_NAME build postgres conjur cli5 cuke-master
```
[the remark this answers is not in the document]
Comment (BradleyBoutcher) [reply bin/start-conjur.sh:20]: We don't pull or build cuke in the oss-only case, but this implementation is cleaner so I'll add it

File: docker-compose.yml (hunk 1 above)
Comment (BradleyBoutcher) [reply docker-compose.yml:30]: I think it's a small enough addition that we can keep them in the same file, if we were gonna do more than two go versions I think I'd definitely want to use the other method you suggest here

File: bin/start-conjur.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (BradleyBoutcher) [reply bin/start-conjur.sh:20]: Good catch, this is much cleaner, thanks!

**Review by diverdane — APPROVED** [review]
LGTM!

**Pull request merged**